src/interactions-tests/utils.ts

```diff
@@ -0,0 +1,213 @@
+import { within, userEvent, fireEvent, queries } from "@storybook/testing-library";
+import { Screen, waitFor, BoundFunctions } from "@testing-library/react";
+import { NavigationCommand } from "./constants";
+import { expect } from "@storybook/jest";
+export type TestFunction = (canvas: BoundFunctions<typeof queries>, args: Record<string, any>) => unknown;
+export type Coordinates = { x: number; y: number };
+
+// Internal functions
+const testFunctionWrapper = (testFunc: TestFunction) => {
+  return async ({ canvasElement, args }: { canvasElement: Screen; args: Record<string, any> }) => {
+    // Starts querying the component from its root element
+    const canvas = getWithin(canvasElement);
+    return testFunc(canvas, args);
+  };
+};
+
+function logFunctionStart(name: string) {
+  expect(` ➡️ ${name}`).toBeDefined();
+}
+
+function getElementClientCenter(element: HTMLElement) {
+  const { left, top, width, height } = element.getBoundingClientRect();
+  return {
+    x: left + width / 2,
+    y: top + height / 2
+  };
+}
+
+const getCoords = ({
+  toElm,
+  toCoords,
+  delta,
+  from
+}: {
+  toElm: HTMLElement;
+  toCoords: Coordinates;
+  delta: Coordinates;
+  from: Coordinates;
+}) => {
+  if (toCoords) {
+    return { ...from, ...toCoords };
+  }
+  if (toElm) {
+    return getElementClientCenter(toElm);
+  }
+  if (delta) {
+    return {
+      x: from.x + delta.x,
+      y: from.y + delta.y
+    };
+  }
+  return {
+    x: from.x + 10,
+    y: from.y + 0
+  };
+};
+
+function getWithin(canvasOrValidTestElement: HTMLElement | BoundFunctions<typeof queries>) {
+  if (canvasOrValidTestElement instanceof HTMLElement) {
+    const result = within(canvasOrValidTestElement);
+    if (result instanceof Error) {
+      throw result;
+    }
+    return result;
+  } else if (canvasOrValidTestElement.getByRole) return canvasOrValidTestElement;
+}
+
+// External functions
+export const interactionSuite =
+  ({
+    beforeEach = null,
```

Comment:
as this something that is being exported, i would add `beforeAll` and `afterAll`

and as a general approach i would try to keep it as close to jest standard syntax